code/modules/mob/living/living.dm

```diff
@@ -443,6 +443,11 @@
 	if (s_active && !(s_active.ClickAccessible(src, depth=STORAGE_VIEW_DEPTH) || s_active.Adjacent(src)))
 		s_active.close(src)
 
+	for(var/mob/M in oview(src))
+		M.update_vision_cone()
```

Comment:
instead of updating their vision cone for everyone in their vone, just pass through (src) and calculate if that is in the vision cone hidden area for M, that way you save a little performance